cmd/fileexperts/fileexperts.go

```diff
@@ -0,0 +1,140 @@
+package fileexperts
+
+import (
+	"fmt"
+
+	apicmd "github.com/wakatime/wakatime-cli/cmd/api"
+	paramscmd "github.com/wakatime/wakatime-cli/cmd/params"
+	"github.com/wakatime/wakatime-cli/pkg/apikey"
+	"github.com/wakatime/wakatime-cli/pkg/exitcode"
+	"github.com/wakatime/wakatime-cli/pkg/fileexperts"
+	"github.com/wakatime/wakatime-cli/pkg/filter"
+	"github.com/wakatime/wakatime-cli/pkg/heartbeat"
+	"github.com/wakatime/wakatime-cli/pkg/log"
+	"github.com/wakatime/wakatime-cli/pkg/project"
+	"github.com/wakatime/wakatime-cli/pkg/vipertools"
+	"github.com/wakatime/wakatime-cli/pkg/wakaerror"
+
+	"github.com/spf13/viper"
+)
+
+// Params contains file-expert command parameters.
+type Params struct {
+	Entity string
+	Params paramscmd.Params
+}
+
+// Run executes the file-experts command.
+func Run(v *viper.Viper) (int, error) {
+	output, err := FileExperts(v)
+	if err != nil {
+		if errwaka, ok := err.(wakaerror.Error); ok {
+			return errwaka.ExitCode(), fmt.Errorf("file experts fetch failed: %s", errwaka.Message())
+		}
+
+		return exitcode.ErrGeneric, fmt.Errorf(
+			"file experts fetch failed: %s",
+			err,
+		)
+	}
+
+	log.Debugln("successfully fetched file experts")
+	fmt.Println(output)
+
+	return exitcode.Success, nil
+}
+
+// FileExperts returns a rendered file experts of todays coding activity.
+func FileExperts(v *viper.Viper) (string, error) {
+	params, err := LoadParams(v)
+	if err != nil {
+		return "", err
+	}
+
+	handleOpts := initHandleOptions(params.Params)
+
+	apiClient, err := apicmd.NewClientWithoutAuth(params.Params.API)
+	if err != nil {
+		return "", fmt.Errorf("failed to initialize api client: %w", err)
+	}
+
+	handle := fileexperts.NewHandle(apiClient, handleOpts...)
+
+	results, err := handle([]heartbeat.Heartbeat{{Entity: params.Entity}})
+	if err != nil {
+		return "", err
+	}
+
+	if len(results) == 0 {
+		return "", nil
+	}
+
+	output, err := fileexperts.RenderFileExperts(
+		results[0].FileExpert.(*fileexperts.FileExperts),
+		params.Params.StatusBar.Output,
+	)
+	if err != nil {
+		return "", fmt.Errorf("failed generating fileexpert output: %s", err)
+	}
+
+	return output, nil
+}
+
+// LoadParams loads file-expert config params from viper.Viper instance. Returns ErrAuth
+// if failed to retrieve api key.
+func LoadParams(v *viper.Viper) (Params, error) {
+	if v == nil {
+		return Params{}, fmt.Errorf("viper instance is nil")
+	}
+
+	if !v.IsSet("file-experts") {
+		return Params{}, fmt.Errorf("file-experts unset")
+	}
+
+	// it's a hack to get entity working with fileexpert
+	v.SetDefault("entity", vipertools.GetString(v, "file-experts"))
```

Comment:
We always want to require `--entity` when we have `--file-experts` and make `--file-experts` a boolean flag not a string cmd.